### ISSUE TITLE
search: empty search not including data records

### ISSUE THREAD

**Issue by tiborsimko** [cernopendata/opendata.cern.ch#3507]
**Current behaviour**

As a developer, when I load the site partially, e.g. only a few records and one doc, then the empty search seems to be giving only doc hits, not data records hits.

How to reproduce:

First, build and run an "empty" portal:

```
$ git log -1 --format=oneline
69553f6d6dcc3871cd84bce9a9cdff339d4de103 (HEAD -> master, upstream/pr/3504, upstream/master, origin/master) Merge branch 'qa'
$ sudo git clean -d -ff -x
$ podman-compose --podman-build-args='--format docker' down -v
$ podman-compose --podman-build-args='--format docker' -f docker-compose-dev.yml down -v
$ podman-compose --podman-build-args='--format docker' build
$ podman-compose --podman-build-args='--format docker' -f docker-compose-dev.yml build
$ podman-compose --podman-build-args='--format docker' -f docker-compose-dev.yml up -d
$ podman exec -i -t opendatacernch_web_1 ./scripts/populate-instance.sh --skip-docs --skip-glossary --skip-records
```

Second, let's load some data records and check the search:

```
$ podman exec -i -t opendatacernch_web_1 cernopendata fixtures records --mode insert -f cernopendata/modules/fixtures/data/records/cms-primary-datasets.json
$ firefox http://127.0.0.1:5000/search  # seeing 1 doc hit only
```

This leads to seeing a message _"Oops! Something went wrong while fetching results."_ and a "Bad request" status in the logs:

```
/opt/invenio/var/instance/python/lib/python3.9/site-packages/marshmallow/fields.py:944: RemovedInMarshmallow4Warning: Passing field metadata as keyword arguments is deprecated. Use the explicit `metadata=...` argument instead. Additional metadata: {'load_from': 'from'}
  super().__init__(**kwargs)

/opt/invenio/var/instance/python/lib/python3.9/site-packages/marshmallow/fields.py:944: RemovedInMarshmallow4Warning: The 'missing' argument to fields is deprecated. Use 'load_default' instead.
  super().__init__(**kwargs)

[opensearch]            | [2023-12-21T16:41:04,785][DEPRECATION][o.o.d.s.a.I.Parser       ] [cca539b1e62f] Deprecated aggregation order key [_term] used, replaced by [_key]
10.89.1.176 - - [21/Dec/2023 16:41:04] "GET /api/records/?q=&sort=-mostrecent&page=1&size=10 HTTP/1.1" 400 -
```

Third, let's load some docs records:

```
$ podman exec -i -t opendatacernch_web_1 cernopendata fixtures docs --mode insert -f /code/cernopendata/modules/fixtures/data/docs/lhcb-releases-entire-run1-dataset-2023/lhcb-releases-entire-run1-dataset-2023.json
$ firefox http://127.0.0.1:5000/search
```

This leads to seeing one hit (the docs record).

**Expected behaviour**

The empty search should give both docs records and data records.

**Notes**

Note that non-empty searches seem to work well, e.g. searching for "cms" finds well 26 data records:

```
$ firefox 'http://127.0.0.1:5000/search?q=cms'
```

**Comment by psaiz** [comment]
This looks like a cache issue. Note that, by default, opensearch caches the queries. A call like 

curl https://127.0.0.1:9200/_cache/clear 

might be enough to do the trick

**Comment by tiborsimko** [comment]
> This looks like a cache issue. Note that, by default, opensearch caches the queries.

Can we make this behaviour configurable by respecting DEBUG=True or some such environment variable, so that in development mode the search cache would not be used by default at all? We could then set different values of the `DEBUG` environment variable in `docker-compose.yml` and `docker-compose-dev.yml`. This would make for a more comfortable local developing experience.

**Comment by tiborsimko** [comment]
> This looks like a cache issue.

BTW, it is not only related cache, because when I don't load any docs (`/scripts/populate-instance.sh --skip-docs --skip-glossary --skip-record`), only some records (`cernopendata fixtures records --mode insert-or-replace -f cernopendata/modules/fixtures/data/records/cms-primary-datasets.json`),  the empty search gape leads to the "oops" messages, and flushing the cache (`curl -k -X POST https://127.0.0.1:9200/_cache/clear`) is not helping.

**Comment by psaiz** [comment]
This was related to the default sorting, by `newest`, and the fact that only the docs had the field that was used for that sorting. We have modified the templates, and now the records and glosssary also include that field. Do you still see the issue?

**Comment by tiborsimko** [comment]
Yes, the problem is now solved on the latest `master` branch, so we can close this issue.

(It would be good if you could always add `Closes #3507` note to the commit message body of the fixing pull request, so that these issues are auto-closed automatically, and so that we keep a reference from the issue to the fix and vice versa.)
